### PR TITLE
Conner/ds api fetch cities

### PR DESCRIPTION
**Pull request by ghost**
With the release of the DS API we no longer need cities.csv and parseCities(), as all locations can be obtained through a simple GET request.

This PR removes legacy code regarding the csv file and replaces it with a new helper method for fetching all cities, titled fetchAllCities(). I have also modified VC code to fetch the cities from the API and display them.

### CODE REVIEW

File: labs-ios-starter/Helpers/DSAPIHelpers.swift [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            NSLog("fetchAllCities(): error: \(error)")
+            completion([])
+            return
+        }
+        
+        if let response = response as? HTTPURLResponse {
+            if response.statusCode == 200 {
+                
+                guard let data = data else {
+                    NSLog("fetchAllCities(): API Response 200, but no data received.")
```
Comment (clc80) [inline labs-ios-starter/Helpers/DSAPIHelpers.swift:34]: I like your log string responses, makes what is happening very clear.

File: labs-ios-starter/Cities/CitiesViewController.swift [hunk 2]
```diff
@@ -30,10 +35,8 @@ extension CitiesViewController: UICollectionViewDelegate, UICollectionViewDataSo
         let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "cityCell", for: indexPath) as! CityCollectionViewCell
         
         let city: String = citySearchResults[indexPath.row]
-        let cityName: String = String(city.dropLast(3))
-        let state: String = String(city.dropFirst(city.count-2))
 
-        cell.cityStateLabel.text = "\(cityName), \(state)"
+        cell.cityStateLabel.text = city
         cell.populationLabel.text = "Population: 2.7M"
```
Comment (clc80) [inline labs-ios-starter/Cities/CitiesViewController.swift:40]: Maybe we should get rid of these for now and add then when and if we have the other data.

**Review by clc80 — APPROVED** [review]

**Review by matt-martindale — APPROVED** [review]
Looks good

File: labs-ios-starter/Cities/CitiesViewController.swift (hunk 2 above)
Comment (ghost) [reply labs-ios-starter/Cities/CitiesViewController.swift:40]: I think i'll remove the populationLabel altogether, good catch

**Pull request merged**